core/src/main/java/org/apache/iceberg/BaseUpdatePartitionSpec.java

```diff
@@ -133,6 +133,11 @@ public BaseUpdatePartitionSpec addField(String name, Term term) {
     Pair<Integer, String> validationKey = Pair.of(sourceTransform.first(), sourceTransform.second().toString());
 
     PartitionField existing = transformToField.get(validationKey);
+    if (existing != null && deletes.contains(existing.fieldId()) &&
+        existing.transform().toString().equals(sourceTransform.second().toString())) {
```

Comment:
Do we need to compare the `toString` representation? I thought everything implemented `equals` correctly.

Reply:
updated